container-system.nix

```diff
@@ -21,23 +27,33 @@ let
       externalInterface = "eth0";
     };
   };
-  nixos = import <nixpkgs/nixos> {
+  nixos = if lib ? nixosSystem
+  then lib.nixosSystem {
+    modules = [ configuration ];
+    system = pkgs.system;
+  }
+  else import "${pkgs.path}/nixos" {
     inherit configuration;
-    system = builtins.currentSystem;
+    system = pkgs.system;
   };
-  system = nixos.config.system.build.toplevel;
-  # older version of nixos-container is required for compatibility with older systemd
-  nixos-container =
-    (import (builtins.fetchTarball https://github.com/NixOS/nixpkgs-channels/archive/nixos-19.09.tar.gz) {})
-    .nixos-container;
+  system = nixos.config.environment.etc."systemd/system".source;
+
+  install-nixos-container = pkgs.writeShellScript "" ''
+    set -euo pipefail
+    ln -is ${system}/{nat,container@}.service /etc/systemd/system
+    mkdir -p /etc/systemd/system/network.target.wants
+    ln -is ${system}/nat.service /etc/systemd/system/network.target.wants
+    systemctl daemon-reload
+  '';
 in
-  { pkgs ? import <nixpkgs> {} }: with pkgs;
+  with pkgs;
   stdenv.mkDerivation {
     name = "container-system";
     unpackPhase = ":";
     installPhase = ''
       mkdir -p $out/bin $out/etc/systemd/system
       ln -s ${nixos-container}/bin/nixos-container $out/bin/nixos-container
-      ln -s ${system}/etc/systemd/system/{nat,container@}.service $out/etc/systemd/system/
+      ln -s ${system}/{nat,container@}.service $out/etc/systemd/system/
```

Comment:
Why did the source path change here?

Do we even need these links any more, now that they are in the install script?

Reply:
It was to prevent building the unneeded parts of NixOS, and only build the `/etc/systemd/system` derivation

Reply:
> Do we even need these links any more, now that they are in the install script?

To install manually if the install script doesn't work for some reason. I guess it can be removed as well, whichever you can prefer.

Reply:
I don't have a strong preference on the links.  I didn't notice you updated `system` to just be the systemd config.  That's cool.